src/rpc/rawtransaction.cpp

```diff
@@ -2042,48 +2047,57 @@ static RPCHelpMan createpsbt()
         throw JSONRPCError(RPC_INVALID_PARAMETER, "The PSBT version can only be 2 or 0");
     }
 
-    PartiallySignedTransaction psbtx(rawTx, psbt_version);
-/*
-    for (unsigned int i = 0; i < rawTx.vout.size(); ++i) {
-        psbtx.outputs[i].blinding_pubkey = output_pubkeys[i];
-    }
-
-    // Add peg-in stuff if it's there
+    // Make a blank psbt
+    std::set<uint256> new_assets;
+    std::set<uint256> new_reissuance;
     for (unsigned int i = 0; i < rawTx.vin.size(); ++i) {
-        if (psbtx.tx->vin[i].m_is_pegin) {
-            CScriptWitness& pegin_witness = psbtx.tx->witness.vtxinwit[i].m_pegin_witness;
-            CAmount val;
-            VectorReader vr_val(SER_NETWORK, PROTOCOL_VERSION, pegin_witness.stack[0], 0);
-            vr_val >> val;
-            psbtx.inputs[i].value = val;
-            VectorReader vr_asset(SER_NETWORK, PROTOCOL_VERSION, pegin_witness.stack[1], 0);
-            vr_asset >> psbtx.inputs[i].asset;
-            VectorReader vr_genesis(SER_NETWORK, PROTOCOL_VERSION, pegin_witness.stack[2], 0);
-            vr_genesis >> psbtx.inputs[i].genesis_hash;
-            psbtx.inputs[i].claim_script.assign(pegin_witness.stack[3].begin(), pegin_witness.stack[3].end());
-
-            VectorReader vr_tx(SER_NETWORK, PROTOCOL_VERSION, pegin_witness.stack[4], 0);
-            VectorReader vr_proof(SER_NETWORK, PROTOCOL_VERSION, pegin_witness.stack[5], 0);
-            if (Params().GetConsensus().ParentChainHasPow()) {
-                Sidechain::Bitcoin::CTransactionRef tx_btc;
-                vr_tx >> tx_btc;
-                psbtx.inputs[i].peg_in_tx = tx_btc;
-                Sidechain::Bitcoin::CMerkleBlock tx_proof;
-                vr_proof >> tx_proof;
-                psbtx.inputs[i].txout_proof = tx_proof;
+        if (!rawTx.vin[i].assetIssuance.IsNull()) {
+            const UniValue& blind_reissuance_v = find_value(request.params[0].get_array()[i].get_obj(), "blind_reissuance");
```

Comment:
We should probably have an `else` on this if statement where we also check this field, and if it's present, we should return an error rather than silently failing to issue the asset.

Reply:
Should check all the issuance-related fields*

Reply:
Done in `ConstructTransaction`.